core-processor/src/ext.rs

```diff
@@ -617,7 +617,7 @@ impl<LP: LazyPagesInterface> ProcessorExternalities for Ext<LP> {
             gas_amount: gas_counter.to_amount(),
             gas_reserver,
             system_reservation_context,
-            allocations: (allocations != initial_allocations).then_some(allocations),
+            allocations: allocations_changed.then_some(allocations),
```

Comment:
Please append a comment here that allocations_changed can be some times true event if final state of allocations is the same as before execution. `allocations_changed` allocations has been changed at least once per execution.